src/harness/BenchmarkDotNet.Extensions/ValuesGenerator.cs

```diff
@@ -182,5 +184,23 @@ private static Guid GenerateRandomGuid(Random random)
             random.NextBytes(bytes);
             return new Guid(bytes);
         }
+
+        private static decimal GenerateRandomDecimal(Random random)
+        {
+            byte scale = (byte)random.Next(29);
+            bool sign = random.Next(2) == 0;
+            return new decimal(random.NextInt32(),
+                               random.NextInt32(),
+                               random.NextInt32(),
+                               sign,
+                               scale);
+        }
+
+        private static int NextInt32(this Random random)
```

Comment:
Per Jon Skeet, the `Random.Next(int)` method will never return a _negative_ value, those eliminating an entire half of the bit-space of an `int`. This code will return an integer across the entire `int` value set.

Reply:
Could you please add a comment to this method explaining what it does and why we can't start using it in `GenerateValue<T>` now?

Reply:
Done